includes/payment-method/class-giropay.php

```diff
@@ -0,0 +1,284 @@
+			$status = $intent->get_status();
+			$amount = $order->get_total();
+
+			if ( 'succeeded' === $status ) {
+				$transaction_url = $this->compose_transaction_url( $intent->get_charge_id() );
+				$note            = sprintf(
+					WC_Payments_Utils::esc_interpolated_html(
+						/* translators: %1: the successfully charged amount, %2: transaction ID of the payment */
+						__( 'A payment of %1$s was <strong>successfully charged</strong> using WooCommerce Payments (<a>%2$s</a>).', 'woocommerce-payments' ),
+						[
+							'strong' => '<strong>',
+							'a'      => ! empty( $transaction_url ) ? '<a href="' . $transaction_url . '" target="_blank" rel="noopener noreferrer">' : '<code>',
+						]
+					),
+					wc_price( $amount ),
+					$intent_id
+				);
+				$order->add_order_note( $note );
+				$order->payment_complete( $intent_id );
+				return;
+			} else {
+				$error = $intent->get_last_payment_error();
+				if ( ! empty( $error ) ) {
+					$error_message = isset( $error['message'] ) ? $error['message'] : 'unknown';
+					$error_code    = isset( $error['code'] ) ? $error['code'] : 'unknown';
+					Logger::log( sprintf( 'Giropay failed: %s (%s)', $error_message, $error_code ) );
+					/* translators: localized exception message */
+					$order->update_status( 'failed', sprintf( __( 'Giropay payment failed: %s', 'woocommerce-payments' ), $error_message ) );
+				}
+
+				wc_add_notice( __( 'Giropay payment has failed. If you continue to see this notice, please contact the admin.', 'woocommerce-payments' ), 'error' );
+				wp_safe_redirect( wc_get_checkout_url() );
+				exit;
+			}
+		} catch ( Exception $e ) {
+			Logger::log( 'Error: ' . $e->getMessage() );
+
+			/* translators: localized exception message */
+			$order->update_status( 'failed', sprintf( __( 'Giropay payment failed: %s', 'woocommerce-payments' ), $e->getLocalizedMessage() ) );
+
+			wc_add_notice( $e->getLocalizedMessage(), 'error' );
+			wp_safe_redirect( wc_get_checkout_url() );
+			exit;
+		}
+	}
+
+	/**
+	 * Check for a redirect payment method on order received page.
+	 */
+	public function maybe_process_redirect_order() {
+		if ( ! is_order_received_page() || empty( $_GET['payment_intent_client_secret'] ) || empty( $_GET['payment_intent'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification
+			return;
+		}
+
+		$order_id = isset( $_GET['order_id'] ) ? wc_clean( wp_unslash( $_GET['order_id'] ) ) : ''; // phpcs:ignore WordPress.Security.NonceVerification
+
+		$this->process_redirect_payment( $order_id );
+	}
+
+	/**
+	 * Renders the Credit Card input fields needed to get the user's payment information on the checkout page.
+	 *
+	 * We also add the JavaScript which drives the UI.
+	 */
+	public function payment_fields() {
+		try {
+			wp_localize_script( 'wcpay-giropay-checkout', 'wcpay_config', $this->get_payment_fields_js_config() );
+			wp_enqueue_script( 'wcpay-giropay-checkout' );
+
+			wp_enqueue_style(
+				'wcpay-giropay-checkout',
```

Comment:
Probably we should use the same handle as card. `wcpay-checkout`. As it is we're embeding the same css file 3 times.
```
<link rel='stylesheet' id='wcpay-checkout-css'  href='https://dev.nnucomputerwhiz.com/plugins/woocommerce-payments/dist/checkout.css?ver=2.2.0' media='all' />
<link rel='stylesheet' id='wcpay-sepa-checkout-css'  href='https://dev.nnucomputerwhiz.com/plugins/woocommerce-payments/dist/checkout.css?ver=2.2.0' media='all' />
<link rel='stylesheet' id='wcpay-giropay-checkout-css'  href='https://dev.nnucomputerwhiz.com/plugins/woocommerce-payments/dist/checkout.css?ver=2.2.0' media='all' />
```
SEPA has the some problem. 

`wcpay_config` var is also duplicated. Probably we should pull these out into a separate protected method in the base class and call it it way so that it's only triggered once if 3 methods are enabled. We could put this optimization in a separate PR.

Reply:
Great, thanks for this! I renamed the style to avoid that duplication and [created this issue](https://github.com/Automattic/woocommerce-payments/issues/1598) for the `wcpay_config` optimization.